### PR TITLE
gcbmgr: add list build jobs

### PR DESCRIPTION
**Pull request by cpanato**
### What type of PR is this?
This PR adds the option to list the last X build jobs
example:

```
$ krel gcbmgr --branch release-1.18 --list-jobs 8
INFO Running gcbmgr with the following options: {false false false release-1.18 prerelease   8}
INFO Build options: {  cloudbuild.yaml   kubernetes-release-test false false false   }
INFO Retrieving Kubernetes release version for ci/latest on branch release-1.18
INFO Retrieving Kubernetes build version from https://dl.k8s.io/ci/latest-1.18.txt...
INFO Retrieved Kubernetes version: v1.18.0-rc.1.1+d5444327df0357
INFO Trying to get the kube-cross version for release-1.18...
INFO Found the following kube-cross version: v1.13.8-1
INFO Listing GCB substitutions prior to build submission...
INFO RC_TAG:
INFO RELEASE_BRANCH: release-1.18
INFO BUILDVERSION: --buildversion=v1.18.0-rc.1.1+d5444327df0357
INFO KUBE_CROSS_VERSION: v1.13.8-1
INFO TOOL_ORG: kubernetes
INFO TOOL_REPO: release
INFO OFFICIAL_TAG:
INFO OFFICIAL:
INFO RC:
INFO TOOL_BRANCH: master
INFO BUILD_POINT: v1.18.0-rc.1.1-d5444327df0357
INFO NOMOCK_TAG:
INFO NOMOCK:
INFO GCP_USER_TAG: ctadeu-at-gmail-com
INFO Listing last 8 GCB jobs:
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+
|           START TIME           |          FINISH TIME           |  STATUS   |                                                 CONSOLE LOGS                                                  |
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+
| 2020-03-17T18:01:39.454011569Z | 2020-03-17T18:18:42.306307Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/66caa764-d21b-465d-95d6-d8b754ed945b?project=648026197307 |
| 2020-03-17T17:05:46.596215978Z | 2020-03-17T17:58:40.563946Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/22413ad5-0e74-4118-98a8-c6a6b776f2e5?project=648026197307 |
| 2020-03-17T16:44:14.493028495Z | 2020-03-17T17:01:08.357092Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/313eaeba-386b-4744-ac5e-de2da0c0ec3a?project=648026197307 |
| 2020-03-17T15:47:22.036533888Z | 2020-03-17T16:39:33.880910Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/f7fb4817-0d2e-40ae-9fed-ec2f4ba90bb6?project=648026197307 |
| 2020-03-17T15:46:59.446635185Z | 2020-03-17T15:48:00.752557291Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/9283b1b4-e23c-4160-89d1-c9ca7e7cd326?project=648026197307 |
| 2020-03-17T15:41:08.534867366Z | 2020-03-17T15:48:10.707409281Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/afba1c94-a3b1-4a9f-a025-79549f5bf5f4?project=648026197307 |
| 2020-03-12T23:13:39.983798382Z | 2020-03-12T23:42:05.867161Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/66715bd8-93a2-4000-86e9-ab734b72766a?project=648026197307 |
| 2020-03-12T22:47:12.056531535Z | 2020-03-12T22:47:23.343306181Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/88acc767-adfe-4ea4-8646-0287b05d5410?project=648026197307 |
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+
```
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
gcbmgr: Add list build jobs
```

### CODE REVIEW

**Comment by cpanato** [comment]
the test is an initial one, we can discuss how to improve it.

File: pkg/gcp/build/build.go [hunk 1]
```diff
@@ -354,3 +358,28 @@ func mergeMaps(maps ...map[string]string) map[string]string {
 	}
 	return out
 }
+
+func ListJobs(project string, lastJobs int64) error {
+	ctx := context.Background()
+	opts := option.WithCredentialsFile("")
```
Comment (saschagrunert) [inline pkg/gcp/build/build.go]: As a follow up I suggest to wrap this in a new struct like we do in the `guthub` package. :)

File: pkg/gcp/build/build.go [hunk 2]
```diff
@@ -354,3 +358,28 @@ func mergeMaps(maps ...map[string]string) map[string]string {
...
+
+	service, err := cloudbuild.NewService(ctx, opts)
+	if err != nil {
+		return errors.Wrap(err, "error fetching gcloud credentials... try running \"gcloud auth application-default login\"")
+	}
+
+	req, err := service.Projects.Builds.List(project).PageSize(lastJobs).Do()
```
Comment (saschagrunert) [inline pkg/gcp/build/build.go]: Is there a maximum value for the page size? If yes, what do we do if we’re requesting more jobs?

File: pkg/gcp/build/build.go [hunk 3]
```diff
@@ -354,3 +358,28 @@ func mergeMaps(maps ...map[string]string) map[string]string {
...
+	if err != nil {
+		return errors.Wrap(err, "error listing the builds")
```
Comment (saschagrunert) [inline pkg/gcp/build/build.go]: Nit: let’s avoid the word “error” in error messages. Usually the user gets a `logrus.Fatalf` which already indicates that it’s an error.

File: pkg/gcp/build/build.go (hunk 1 above)
Comment (cpanato) [reply pkg/gcp/build/build.go]: ok, I will review the github package and apply the same in a follow up.
thanks!

File: pkg/gcp/build/build.go (hunk 2 above)
Comment (cpanato) [reply pkg/gcp/build/build.go]: i tried to find in the docs but it does not say anything
https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.builds/list

File: pkg/gcp/build/build.go (hunk 3 above)
Comment (cpanato) [reply pkg/gcp/build/build.go]: done, thanks for the review!

File: pkg/gcp/build/build.go (hunk 2 above)
Comment (saschagrunert) [reply pkg/gcp/build/build.go]: Fair, do page sizes > 100 work?

Comment (cpanato) [reply pkg/gcp/build/build.go]: i did with 301 and returned all 301 last jobs

Comment (saschagrunert) [reply pkg/gcp/build/build.go]: Sounds good to me, thanks for verifying.

Comment (cpanato) [reply pkg/gcp/build/build.go]: tried with 500 :)

Comment (cpanato) [reply pkg/gcp/build/build.go]: i think this is not useful but a user can do weird things 😂

Comment (saschagrunert) [reply pkg/gcp/build/build.go]: What happens on `-1`?

Comment (cpanato) [reply pkg/gcp/build/build.go]: it brings the jobs in the future :P 

but looks like it not fail
```
INFO Listing last -1 GCB jobs:
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+
|           START TIME           |          FINISH TIME           |  STATUS   |                                                 CONSOLE LOGS                                                  |
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+
| 2020-03-17T18:01:39.454011569Z | 2020-03-17T18:18:42.306307Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/66caa764-d21b-465d-95d6-d8b754ed945b?project=648026197307 |
| 2020-03-17T17:05:46.596215978Z | 2020-03-17T17:58:40.563946Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/22413ad5-0e74-4118-98a8-c6a6b776f2e5?project=648026197307 |
| 2020-03-17T16:44:14.493028495Z | 2020-03-17T17:01:08.357092Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/313eaeba-386b-4744-ac5e-de2da0c0ec3a?project=648026197307 |
| 2020-03-17T15:47:22.036533888Z | 2020-03-17T16:39:33.880910Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/f7fb4817-0d2e-40ae-9fed-ec2f4ba90bb6?project=648026197307 |
| 2020-03-17T15:46:59.446635185Z | 2020-03-17T15:48:00.752557291Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/9283b1b4-e23c-4160-89d1-c9ca7e7cd326?project=648026197307 |
| 2020-03-17T15:41:08.534867366Z | 2020-03-17T15:48:10.707409281Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/afba1c94-a3b1-4a9f-a025-79549f5bf5f4?project=648026197307 |
| 2020-03-12T23:13:39.983798382Z | 2020-03-12T23:42:05.867161Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/66715bd8-93a2-4000-86e9-ab734b72766a?project=648026197307 |
| 2020-03-12T22:47:12.056531535Z | 2020-03-12T22:47:23.343306181Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/88acc767-adfe-4ea4-8646-0287b05d5410?project=648026197307 |
| 2020-03-12T22:47:18.045150178Z | 2020-03-12T23:16:03.417070Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/c0cdf52a-d386-4de8-8eaa-c6c31fe86960?project=648026197307 |
| 2020-03-12T22:46:46.270468739Z | 2020-03-12T23:17:50.062883Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/0c33f75a-fe5d-4800-8c93-4864545fc96b?project=648026197307 |
| 2020-03-12T20:56:59.329664144Z | 2020-03-12T23:09:49.748778Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/56dcfb8d-4997-4d80-a26b-d98dcb16354e?project=648026197307 |
| 2020-03-12T20:52:03.334708890Z | 2020-03-12T22:40:16.286748Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/bd24d66d-671f-480f-b940-7dd8e4b195b2?project=648026197307 |
| 2020-03-12T20:49:17.189666168Z | 2020-03-12T22:42:03.250553Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/730e2680-1117-4ceb-b7ee-244ddfc73143?project=648026197307 |
| 2020-03-12T20:25:45.541744471Z | 2020-03-12T20:55:39.045461Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/f4553143-76cf-4302-8f8a-eb378d707755?project=648026197307 |
| 2020-03-12T20:19:33.605482117Z | 2020-03-12T20:46:59.117637Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/3cb43a19-5f65-4540-8eb4-ab6a2b44d1ba?project=648026197307 |
| 2020-03-12T20:18:26.464990013Z | 2020-03-12T20:48:07.174950Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/06b6dbbc-8041-47b1-8e21-38d9a62c917a?project=648026197307 |
| 2020-03-12T18:26:12.101181631Z | 2020-03-12T20:11:08.424601Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/53df157a-1a29-45ae-9ef6-2616cf89d0f4?project=648026197307 |
| 2020-03-12T18:25:17.403597155Z | 2020-03-12T18:25:29.064243395Z | CANCELLED | https://console.cloud.google.com/cloud-build/builds/16b647fc-6609-4a3c-a108-615be37b3477?project=648026197307 |
| 2020-03-12T18:21:02.673580173Z | 2020-03-12T20:24:24.667413Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/2a8c78a4-edd6-4769-8d7e-edb6c5b4062b?project=648026197307 |
| 2020-03-12T18:20:59.567154612Z | 2020-03-12T20:16:06.787010Z    | SUCCESS   | https://console.cloud.google.com/cloud-build/builds/27e6b98a-c59c-41e6-8f93-b0089a88c95e?project=648026197307 |
+--------------------------------+--------------------------------+-----------+---------------------------------------------------------------------------------------------------------------+

```

Comment (justaugustus) [reply pkg/gcp/build/build.go]: Follow-up: Normalize (set to default count) any non-positive number. Also, log that we're going to do that

**Comment by justaugustus** [comment]
Great job, @cpanato!
/lgtm
/approve

**Comment by justaugustus** [comment]
@cpanato -- I had a merge party and now this needs a rebase :stuck_out_tongue_closed_eyes:

**Comment by cpanato** [comment]
done @justaugustus

**Comment by cpanato** [comment]
@justaugustus done. PTAL

**Comment by justaugustus** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**